packages/vite/src/node/plugins/worker.ts

```diff
@@ -278,10 +278,11 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
       }
     },
 
-    renderChunk(code) {
-      if (isWorker && code.includes('import.meta.url')) {
-        return code.replace('import.meta.url', 'self.location.href')
+    resolveImportMeta(property) {
```

Comment:
good point to fix `import.meta.url` in worker, but I think this is no part of the PR.

Reply:
I'll revert this, let me know if you want me to make another PR for this update!

Reply:
I actually can't revert this and keep my new tests for the worker support. The old test suite was missing the check for using `new URL` to construct a worker and it fails without this fix.

Reply:
OK， keep it. I open PR to test it 😊

Reply:
This is indeed a very clean solution 👍

Reply:
I'm reverting [6a0384c](https://github.com/vitejs/vite/pull/7837/commits/6a0384c2bfd135224cf0e8d920f07e9f89a5beb6) since it looks like this was resolved upstream by other means?